### PR TITLE
Pressing a track's title plays/pauses track, clicking artist goes to artist page

### PR DESCRIPTION
**Pull request by peterklingelhofer**
These changes allow for a user to click anywhere on a track's title to play or pause the song for any trackgroup (such as [this](https://stream.resonate.coop/artist/13946/release/shut-the-fuck-up-talking-to-me)).

On mobile, it was unclear for users how to play songs, as the play buttons would be numbers. On desktop, when you mouse over the number it becomes a play button, but mobile users do not have `hover` so it's unclear. These changes make it so pressing on a track's title plays the track, and pressing on a different track's title plays that track instead, just the same as clicking the play button on the left for that track. This makes it way easier to change tracks, since the play/pause button is small, this gives users a bigger area where they can click/press.

### CODE REVIEW

File: packages/track-component/index.js
Comment (fgblomqvist) [inline packages/track-component/index.js]: Is the double-click handler still needed? Can it just be put on the same div as the single-click handler if so?

**Review by fgblomqvist — COMMENTED** [review]
Could you also make sure we get the proper cursor: pointer when hovering over the title? We don't currently have it for the play button either I believe (in fact, we're missing it in tons of locations).

File: packages/track-component/index.js
Comment (peterklingelhofer) [reply packages/track-component/index.js]: Nice catch, I addressed this here: https://github.com/resonatecoop/stream/pull/212/commits/3b48df7538b5dfdc790ce987f8cdf95188f23a6f

**Comment by peterklingelhofer** [comment]
> Could you also make sure we get the proper cursor: pointer when hovering over the title? We don't currently have it for the play button either I believe (in fact, we're missing it in tons of locations).

@fgblomqvist, this latest commit, https://github.com/resonatecoop/stream/pull/212/commits/3b48df7538b5dfdc790ce987f8cdf95188f23a6f, addressed this in a lot of places. I also was able to make the title underline on hover as well, as well as for the artist name. The clicking the artist name now sends you to the artist page as well. I had to do the inline margin adjustments because there was a slight gap between the play button and the track title where clicking wouldn't play/pause and the cursor would stop being pointer, so that's why that's in there. Try it out and let me know what you think!

**Comment by peterklingelhofer** [comment]
Latest commit https://github.com/resonatecoop/stream/pull/212/commits/7fea53870bc57d8e1ce3f72141af9b268facd3a9 gives us row and column resize for volume controller and seeker. However, I'm adding this to the tachyons css itself. Would it be better to add it elsewhere in case we need to update tachyons (per https://github.com/resonatecoop/stream/issues/101)?

File: packages/player-component/index.js
Comment (fgblomqvist) [inline packages/player-component/index.js]: Could this onclick just sit on the span?

File: packages/tachyons/src/modules/_rangeSlider.css
Comment (fgblomqvist) [inline packages/tachyons/src/modules/_rangeSlider.css:13]: You also need to set this on the actual handle, same with the vertical one. The handle has cursor: pointer atm which has higher specificity than this rule.

**Comment by fgblomqvist** [comment]
I'm not a designer, but from a design perspective, I feel like maybe having the row vaguely lit up would look better than the underline. Just my 2 cents though. This is not bad though.

File: packages/player-component/index.js
Comment (fgblomqvist) [inline packages/player-component/index.js]: This div can be removed now 🙂  (otherwise everything looks good now)

**Comment by peterklingelhofer** [comment]
> I'm not a designer, but from a design perspective, I feel like maybe having the row vaguely lit up would look better than the underline. Just my 2 cents though. This is not bad though.

I looked into this, and I totally agree. Unfortunately, the Tachyons `glow` and `glow-large` classes don't seem to have any effect for dark mode users. Maybe in a subsequent PR we can iterate on this one and add a hover glow that works for all themes and replace all of the `underline-hover` usages with that.

**Review by fgblomqvist — APPROVED** [review]

**Comment by auggod** [comment]
I don't use cursor:pointer on buttons because buttons are not links. Instead, we should focus on improving the buttons affordance. I can't merge this pull requests as long as 'cursor' is being used on buttons.

**Comment by fgblomqvist** [comment]
> I don't use cursor:pointer on buttons because buttons are not links

But none of these buttons actually look like buttons? They're just arbitrary elements that utilize the button element but completely different styling than what makes a button a button. The argument, afaik, is that buttons don't need cursor pointer because they look like buttons, compared to links that don't.

E.g. what makes the play icon look like a button? What makes the 9 dots not look like a button?

Or is there some accessibility docs on this that I might have missed?

**Comment by auggod** [comment]
I agree there's a lack of affordance if you compare the the play count dots and the track menu button. The track menu button does have the grow class but the effect is too subtle.

There's this article on medium about this
https://medium.com/simple-human/buttons-shouldnt-have-a-hand-cursor-b11e99ca374b

**Comment by fgblomqvist** [comment]
I think the author is a bit dated in how he thinks. Just like how famous dictionaries get shaped by how society use a language, UX guidelines should be shaped by what society expects. It's not always as formal as one might wish, but it does aid society in the right way.

For example, he argues that there's plenty of things on Googles homepage that doesn't have the pointer, and idk if things have changed since, but if you take a look today there is not a single clickable element on their homepage that doesn't have the pointer. The textbox certainly has the text cursor, but that makes sense since it's not "clickable", it's editable.

If you surveyed a thousand random web users of varying backgrounds and ages, and asked them "what does it mean when the cursor turns into a pointer when you hover over something?" do you think they would say "it means it's a link" or do you think they would say "it means I can click it"? Google seems to agree with the latter, and so does every other website I've just taken a look at.

There's some more discussion/background here:
https://ux.stackexchange.com/questions/105024/why-dont-button-html-elements-have-a-css-cursor-pointer-by-default

Ultimately, I would say the argument about affordance was relevant back when buttons looked like buttons and links looked like links, which hasn't been the general case for probably a decade if not more. But, that's just my 2 cents.

That being said, regardless, I agree that the affordance should be improved, if anything just for stylistic reasons.

**Comment by peterklingelhofer** [comment]
[Tidal](https://tidal.com/) for example uses cursor pointer on their buttons, and they're arguably one of the "more modern" and recent streaming platforms to date. It makes it extra obvious for the user what is clickable and what isn't.

Back in the old days where every button had an outline and users could visually see the edge of where they could click I can see why folks would think that would be unnecessary, but none of the buttons in our player (except for the `LOG IN` and `JOIN NOW` buttons) have edges/outlines, so I do think cursor pointer would indeed be a slight accessibility boon for our users. Just my 2 cents though. 😀

What do you think @auggod, did we convince you? If not, I'm happy to remove the `cursor:pointer` from wherever you're disliking it. Just let me know! 😎

**Comment by peterklingelhofer** [comment]
@auggod removed `pointer` from all buttons. Testing fine for me.

**Comment by peterklingelhofer** [comment]
Recreated here: https://github.com/resonatecoop/stream/pull/276

**Pull request closed without merging**
